test/test_torch.py

```diff
@@ -3889,6 +3889,18 @@ def test_comparison_ops_must_take_bool_output(self):
                        torch.logical_and, torch.logical_or, torch.logical_xor]:
                 self.assertEqual(op(torch.tensor([True]), torch.tensor([False])).dtype, torch.bool)
 
+        def test_out_comparison_ops_type_promotion_and_broadcasting(self):
+            # issue #42660
```

Comment:
I expected significantly more tested verifying nan behavior. Relying on `allclose` tests is nice, but direct testing is always preferable. 
Also, tests that verify producing bool output by default - do we have them already?

Reply:
More tests added below.